### PR TITLE
slow node detection: enable `evict-slow-trend` on `raft-kv2`.

### LINKED ISSUE #6868: [Dynamic Regions] enable slow-trend-scheduler by default
### What did you do?
1、inject one of tikv io hang

### What did you expect to see?
qps can recover within 5mins

### What did you see instead?
qps can not recover within 5mins due to slow-trend-scheduler is disable by default

### What version of PD are you using (`pd-server -V`)?
pd githash：ef81248494d86bb7bc804af5299a113882ac93ec
tikv githash：284ff3a9ca144619f5efaa357d6c287f1c669982

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What problem does this PR solve?

Issue Number: Close #6868, Ref https://github.com/tikv/tikv/issues/15271 and Close https://github.com/tikv/tikv/issues/15267

### What is changed and how does it work?

```commit-message
This pr contains:
+ Enable `evict-slow-trend` scheduler on `raft-kv2` as default.
+ Optimize the detection strategy on the perception of Network I/O delays on TiKV nodes.
```

Records when building a cluster with `raft-kv-2` engine(multi-rocksdb):
```
[2023/08/04 07:33:40.560 +00:00] [INFO] [store_config.go:231] ["sync the store config successful"] [store-address=10.233.108.15:20180] [store-config="{\n  \"coprocessor\": {\n    \"region-max-size\": \"15GiB\",\n    \"region-split-size\": \"10GiB\",\n    \"region-max-keys
\": 153600000,\n    \"region-split-keys\": 102400000,\n    \"enable-region-bucket\": true,\n    \"region-bucket-size\": \"50MiB\"\n  },\n  \"storage\": {\n    \"engine\": \"raft-kv2\"\n  }\n}"] [old-config="{\n  \"coprocessor\": {\n    \"region-max-size\": \"\",\n    \"re
gion-split-size\": \"\",\n    \"region-max-keys\": 0,\n    \"region-split-keys\": 0,\n    \"enable-region-bucket\": false,\n    \"region-bucket-size\": \"\"\n  },\n  \"storage\": {\n    \"engine\": \"\"\n  }\n}"]
[2023/08/04 07:33:40.561 +00:00] [INFO] [cluster.go:440] ["create scheduler"] [scheduler-name=evict-slow-trend-scheduler] [scheduler-args="[]"]
[2023/08/04 07:33:40.561 +00:00] [INFO] [cluster.go:446] ["add scheduler successfully"] [scheduler-name=evict-slow-trend] [scheduler-args="[]"]
```

And we got the following metrics:
image

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/schedulers/evict_slow_trend.go
Comment (rleungx) [inline pkg/schedule/schedulers/evict_slow_trend.go:416]: Why change it?

File: pkg/mcs/scheduling/server/config/config.go
Comment (rleungx) [inline pkg/mcs/scheduling/server/config/config.go:639]: Move it before line 614 since it is implemented.

File: pkg/schedule/schedulers/evict_slow_trend.go
Comment (rleungx) [inline pkg/schedule/schedulers/evict_slow_trend.go:339]: Please change `cand.add` format to be consistent with others.

File: pkg/schedule/schedulers/evict_slow_trend.go
Comment (rleungx) [inline pkg/schedule/schedulers/evict_slow_trend.go:317]: ditto

File: pkg/schedule/schedulers/evict_slow_trend.go
Comment (LykxSassinator) [reply pkg/schedule/schedulers/evict_slow_trend.go:416]: `Before` is not accurate for the judgement so I changed it.

When `slowStore.LastHeartBeatTS == store.LastHeartBeatTS`, it might miss it. However, in this case, it means that the PD simultaneously receives the heartbeats from slow node and other nodes. And it also means the heartbeat is already fresh to make the judgement, no need to wait another heartbeat tick.

File: pkg/schedule/schedulers/evict_slow_trend.go
Comment (rleungx) [inline pkg/schedule/schedulers/evict_slow_trend.go:345]: duplicated?

File: server/cluster/cluster.go
Comment (rleungx) [inline server/cluster/cluster.go:441]: Do we need to persist it?

Comment (LykxSassinator) [reply server/cluster/cluster.go:441]: Actually not, currently, `slow-trend-scheduler` has no extra arguments to persist. So I skipped it here.

File: pkg/schedule/schedulers/evict_slow_trend.go
Comment (rleungx) [inline pkg/schedule/schedulers/evict_slow_trend.go:123]: recoverTS is set to now directly?

**Review by innerr — APPROVED** [review]
LGTM, but it needs a code-owner approval

File: server/cluster/cluster.go
Comment (rleungx) [inline server/cluster/cluster.go:438]: How about letting coordinator to create the scheduler?

Comment (LykxSassinator) [reply server/cluster/cluster.go:438]: FYI, maybe u means https://github.com/tikv/pd/blob/346e7716e2598dfba2db6afa73c3731e15449f49/server/cluster/cluster.go#L783-L784 ?
It's succinct enough for it.

Comment (LykxSassinator) [reply server/cluster/cluster.go]: As discussed off-line, this part has been polished. And the test results show that it's valid.
image

File: pkg/schedule/coordinator.go
Comment (bufferflies) [inline pkg/schedule/coordinator.go:384]: should change line 348

Comment (LykxSassinator) [reply pkg/schedule/coordinator.go:387]: Thx for reminding, done.

File: pkg/schedule/schedulers/metrics.go
Comment (bufferflies) [inline pkg/schedule/schedulers/metrics.go:159]: should the grafana.json  update else?

Comment (LykxSassinator) [reply pkg/schedule/schedulers/metrics.go:159]: Nope, this part should be polished in next pr(we need another pr to tidy all related metrics, to make it more clear to read)

**Review by bufferflies — APPROVED** [review]

File: pkg/schedule/coordinator.go
Comment (nolouch) [inline pkg/schedule/coordinator.go:316]: `pd-ctl` allows users to add or remove a scheduler through commands, this way make pd ctl be invalid. 
consider defining it in here: https://github.com/tikv/pd/blob/346e7716e2598dfba2db6afa73c3731e15449f49/pkg/schedule/config/config.go#L522-L531

but there is no logic like `if raftkv2 then defaultV2, else then defaultV1`, we may need add this logic.

Comment (LykxSassinator) [reply pkg/schedule/coordinator.go:316]: > but there is no logic like if raftkv2 then defaultV2, else then defaultV1, we may need add this logic.
When initializing schedulers, we cannot know whether the cluster is using raft-kv2 or raft-kv1. Only if the cluster loaded the stores with `raft-kv2` later, can it know whether the cluster is using raft-kv2.

File: pkg/schedule/coordinator.go
Comment (nolouch) [inline pkg/schedule/coordinator.go:330]: How do make sure the config already is loaded or synced from tikv. I think if it is reliable, it's also reliable in `c.initSchedulers()`.

Comment (LykxSassinator) [reply pkg/schedule/coordinator.go:330]: Nope, we cannot do it, as the perception on whether this cluster is raft-kv2 or not is triggered by a asynchronous timer in `syncStoreConfig`.

**Review by nolouch — APPROVED** [review]

File: pkg/schedule/schedulers/metrics.go
Comment (JaySon-Huang) [inline pkg/schedule/schedulers/metrics.go:126]: Should it be `uncatalogued`?
https://dictionary.cambridge.org/zhs/%E8%AF%8D%E5%85%B8/%E8%8B%B1%E8%AF%AD/uncatalogued

Comment (LykxSassinator) [reply pkg/schedule/schedulers/metrics.go:126]: Yep, it's a spelling error. I'll tidy and clean up all misleading metrics in the next pr.

**Pull request merged**
